internal/pkg/cli/local_run_test.go

```diff
@@ -511,6 +512,82 @@ func TestLocalRunOpts_Execute(t *testing.T) {
+			inputAppName:  testAppName,
+			inputWkldName: testWkldName,
+			inputEnvName:  testEnvName,
+			setupMocks: func(m *localRunExecuteMocks) {
+				m.ecsLocalClient.EXPECT().TaskDefinition(testAppName, testEnvName, testWkldName).Return(taskDef, nil)
+				m.ssm.EXPECT().GetSecretValue(gomock.Any(), "mysecret").Return("secretvalue", nil)
+				m.ws.EXPECT().ReadWorkloadManifest(testWkldName).Return([]byte(""), nil)
+				m.interpolator.EXPECT().Interpolate("").Return("", nil)
+
+				runCalled := make(chan struct{})
+				stopCalled := make(chan struct{})
+				m.dockerEngine.EXPECT().Run(gomock.Any(), expectedRunPauseArgs).DoAndReturn(func(ctx context.Context, opts *dockerengine.RunOptions) error {
+					close(runCalled)
+					return nil
+				})
+				m.dockerEngine.EXPECT().IsContainerRunning(mockPauseContainerName).DoAndReturn(func(name string) (bool, error) {
+					<-runCalled
+					return true, nil
+				})
+				m.dockerEngine.EXPECT().Run(gomock.Any(), expectedRunFooArgs).DoAndReturn(func(ctx context.Context, opts *dockerengine.RunOptions) error {
+					syscall.Kill(syscall.Getpid(), syscall.SIGINT)
+					<-stopCalled
+					return errors.New("hi")
+				})
+				m.dockerEngine.EXPECT().Run(gomock.Any(), expectedRunBarArgs).Return(nil)
+
+				m.dockerEngine.EXPECT().Stop(expectedRunFooArgs.ContainerName).DoAndReturn(func(id string) error {
+					close(stopCalled)
+					return errors.New("stop foo")
+				})
+				m.dockerEngine.EXPECT().Stop(expectedRunBarArgs.ContainerName).Return(nil)
+				m.dockerEngine.EXPECT().Rm(expectedRunBarArgs.ContainerName).Return(errors.New("rm bar"))
+				m.dockerEngine.EXPECT().Stop(expectedRunPauseArgs.ContainerName).Return(nil)
+				m.dockerEngine.EXPECT().Rm(expectedRunPauseArgs.ContainerName).Return(errors.New("rm stop"))
+			},
+			wantedError: fmt.Errorf("clean up %q: stop: stop foo\nclean up %q: rm: rm bar\nclean up %q: rm: rm stop", expectedRunFooArgs.ContainerName, expectedRunBarArgs.ContainerName, expectedRunPauseArgs.ContainerName),
+		},
+		"handles ctrl-c successfully, errors from running ignored": {
+			inputAppName:  testAppName,
+			inputWkldName: testWkldName,
+			inputEnvName:  testEnvName,
+			setupMocks: func(m *localRunExecuteMocks) {
+				m.ecsLocalClient.EXPECT().TaskDefinition(testAppName, testEnvName, testWkldName).Return(taskDef, nil)
+				m.ssm.EXPECT().GetSecretValue(gomock.Any(), "mysecret").Return("secretvalue", nil)
+				m.ws.EXPECT().ReadWorkloadManifest(testWkldName).Return([]byte(""), nil)
+				m.interpolator.EXPECT().Interpolate("").Return("", nil)
+
+				runCalled := make(chan struct{})
+				stopCalled := make(chan struct{})
+				m.dockerEngine.EXPECT().Run(gomock.Any(), expectedRunPauseArgs).DoAndReturn(func(ctx context.Context, opts *dockerengine.RunOptions) error {
+					close(runCalled)
+					return nil
+				})
+				m.dockerEngine.EXPECT().IsContainerRunning(mockPauseContainerName).DoAndReturn(func(name string) (bool, error) {
+					<-runCalled
+					return true, nil
+				})
```

Comment:
very interesting! Is the use of channels in testcases to ensure that
`m.dockerEngine.EXPECT().Run` should always be executed before `m.dockerEngine.EXPECT().IsContainerRunning`.